tools/autograd/derivatives.yaml

```diff
@@ -2208,3 +2208,6 @@
 
 - name: segment_reduce(Tensor data, str reduce, *, Tensor? lengths=None, Tensor? indices=None, int axis=0, bool unsafe=False, Scalar? initial=None) -> Tensor
   data: _segment_reduce_backward(grad, result, data, reduce, lengths)
+
+- name: pin_memory(Tensor(a) self, Device? device=None) -> Tensor(a)
```

Comment:
pin_memory is a view op?
If it is a no-op in some cases (returning self as is), this is not ok for autograd. You will have to keep this one as composite and use another function to do the pinning that always return different memory.

Reply:
whoops. OK yes will need to fix this. I wonder if any tests gonna catch this...

Reply:
https://github.com/pytorch/pytorch/pull/60286 should once it lands haha
If you want to test this function only, you can give an input that requires grad and is already pinned, that should fail as well.